### PR TITLE
[TechDocs] API implementation how-to guide

### PR DESCRIPTION
**Pull request by emmaindal**
Signed-off-by: Emma Indal <emma.indahl@gmail.com>

## Hey, I just made a Pull Request!

This PR introduces a two step how-to guide on how to implement custom APIs for TechDocs and override them in the App. 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/techdocs/how-to-guides.md [hunk 1]
```diff
@@ -217,3 +217,56 @@ techdocs:
 [beta-migrate-bug]:
 https://github.com/backstage/backstage/issues/new?assignees=&labels=bug&template=bug_template.md&title=[TechDocs]%20Unable%20to%20run%20beta%20migration
 [using-cloud-storage]: ./using-cloud-storage.md
+
+## How to implement your own TechDocs APIs
+
+The TechDocs plugin provides implementations of two primary apis by default, the
```
Comment (bodilb) [inline docs/features/techdocs/how-to-guides.md:223]:
```suggestion
The TechDocs plugin provides implementations of two primary apis by default: the
```

File: docs/features/techdocs/how-to-guides.md [hunk 2]
```diff
@@ -217,3 +217,56 @@ techdocs:
...
+[TechDocsStorageApi](https://github.com/backstage/backstage/blob/55114cfeb7045e3e5eeeaf67546b58964f4adcc7/plugins/techdocs/src/api.ts#L33),
+which is responsible for talking to TechDocs storage to fetch files to render,
+and
+[TechDocsApi](https://github.com/backstage/backstage/blob/55114cfeb7045e3e5eeeaf67546b58964f4adcc7/plugins/techdocs/src/api.ts#L49),
+which is responsible for talking to techdocs-backend.
+
+There may be occasions where you need to implement these two apis yourself, to
```
Comment (bodilb) [inline docs/features/techdocs/how-to-guides.md:230]:
```suggestion
There may be occasions where you need to implement these two APIs yourself, to
```

File: docs/features/techdocs/how-to-guides.md [hunk 3]
```diff
@@ -217,3 +217,56 @@ techdocs:
...
+customize it to your own needs. The purpose of this guide is to walk you through
```
Comment (bodilb) [inline docs/features/techdocs/how-to-guides.md:231]:
```suggestion
There may be occasions where you need to implement these two APIs yourself, to
customize them to your own needs. The purpose of this guide is to walk you through
```

File: docs/features/techdocs/how-to-guides.md [hunk 4]
```diff
@@ -217,3 +217,56 @@ techdocs:
...
+export class TechDocsCustomStorageApi implements TechDocsStorageApi {
+  // your implementation
+}
+
+export class TechDocsCustomApiClient implements TechDocsApi {
+  // your implementation
+}
+```
+
+2. Override the api refs `techdocsStorageApiRef` and `techdocsApiRef` with your
```
Comment (bodilb) [inline docs/features/techdocs/how-to-guides.md:247]:
```suggestion
2. Override the API refs `techdocsStorageApiRef` and `techdocsApiRef` with your
```

**Review by bodilb — APPROVED** [review]
Niiice! 👌  Just a few tiny edits suggested.

File: docs/features/techdocs/how-to-guides.md [hunk 5]
```diff
@@ -217,3 +217,56 @@ techdocs:
 [beta-migrate-bug]:
 https://github.com/backstage/backstage/issues/new?assignees=&labels=bug&template=bug_template.md&title=[TechDocs]%20Unable%20to%20run%20beta%20migration
 [using-cloud-storage]: ./using-cloud-storage.md
+
+## How to implement your own TechDocs APIs
+
+The TechDocs plugin provides implementations of two primary apis by default: the
```
Comment (bodilb) [inline docs/features/techdocs/how-to-guides.md:223]:
```suggestion
The TechDocs plugin provides implementations of two primary APIs by default: the
```

**Review by camilaibs — APPROVED** [review]
It looks pretty clear from my developer perspective too!

**Pull request merged**